// Copyright 2019 The Tensor2Robot Authors.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto2";

package third_party.py.tensor2robot;

message ExtendedTensorSpec {
  // This message allows to (de)serialize tensorspec_utils.ExtendedTensorSpec.
  // Each field has a one to one mapping to the class constructor.
  repeated int32 shape = 1;
  optional int32 dtype = 2;
  optional string name = 3;
  optional bool is_optional = 4;
  optional bool is_extracted = 5;
  optional string data_format = 6;
  optional string dataset_key = 7;
}

message TensorSpecStruct {
  // This message allows to (de)serialize tensorspec_utils.TensorSpecStruct.
  // This structure is essentially an OrderedDict which is therefore
  // serializable through a key: value map.
  map<string, ExtendedTensorSpec> key_value = 1;
}

message T2RAssets {
  optional TensorSpecStruct feature_spec = 1;
  optional TensorSpecStruct label_spec = 2;
  optional int32 global_step = 3;
}
